### PR TITLE
Fix DPP + SyncBN

### PR DESCRIPTION
**Pull request by BloodAxe**
## What does this PR do?

This PR ensures that model is already on correct GPU before applying SyncBN conversion. According to PyTorch [docs](https://pytorch.org/docs/stable/generated/torch.nn.SyncBatchNorm.html?highlight=syncbatchnorm#torch.nn.SyncBatchNorm.convert_sync_batchnorm), SyncBN conversion should be applied to model that is already on target GPU. Not following this requirement will lead to hard-tracing errors of some BN layers kept on CPU. 
Changing the order of `model_to_device` call seems to fix this issue without any side-effects to the rest of the code.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

**Review by awaelchli — COMMENTED** [review]
looks like that's the correct thing to do.
can you apply the same to `ddp_spawn.py`?

**Comment by BloodAxe** [comment]
Greetings! Thanks for quick response. Added same changes to ddp_spawn in my last commit.

**Review by awaelchli — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/ddp.py [hunk 1]
```diff
@@ -241,12 +241,12 @@ def init_ddp_connection(self, global_rank: int, world_size: int) -> None:
             torch_distrib.init_process_group(self.torch_distributed_backend, rank=global_rank, world_size=world_size)
 
     def pre_dispatch(self):
-        if self.sync_batchnorm:
-            self.model = self.configure_sync_batchnorm(self.model)
-
         # move the model to the correct device
         self.model_to_device()
 
+        if self.sync_batchnorm:
```
Comment (tchaton) [inline pytorch_lightning/plugins/training_type/ddp.py:247]:
```suggestion
        # `sync BatchNorm` requires a model on `cuda` device
        if self.sync_batchnorm:
```

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 2]
```diff
@@ -141,12 +141,12 @@ def new_process(self, process_idx, trainer, mp_queue):
         self.dist.rank = self.global_rank
         self.dist.device = self.root_device
 
-        if self.sync_batchnorm:
-            self.model = self.configure_sync_batchnorm(self.model)
-
         # move the model to the correct device
         self.model_to_device()
 
+        if self.sync_batchnorm:
```
Comment (tchaton) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:147]:
```suggestion
        # `sync BatchNorm` requires a model on `cuda` device
        if self.sync_batchnorm:
```

**Review by tchaton — COMMENTED** [review]
LGTM ! Thanks for this catch !
